crates/eth-rpc/src/servers/alchemy_rpc.rs

```diff
@@ -22,8 +30,117 @@ impl<P: Provider + Send + Sync> AlchemyRpc<P> {
 
 #[async_trait]
 impl<P: Provider + Send + Sync + 'static> AlchemyApiServer for AlchemyRpc<P> {
+    async fn token_allowance(
+        &self,
+        contract_address: Address,
+        account_address: Address,
+        spender_address: Address,
+    ) -> Result<TokenAllowance> {
+        let entry_point: Felt252Wrapper = keccak256("allowance(account, spender)")
+            .try_into()
+            .map_err(|e: ConversionError<()>| EthApiError::<P::Error>::ConversionError(e.to_string()))?;
+        let entry_point: FieldElement = entry_point.into();
+
+        let account_addr: Felt252Wrapper = account_address.into();
+        let account_addr: FieldElement = account_addr.into();
+
+        let spender_addr: Felt252Wrapper = spender_address.into();
+        let spender_addr: FieldElement = spender_addr.into();
+
+        let calldata = vec![entry_point, account_addr, spender_addr];
+
+        let handle = self
+            .kakarot_client
+            .call(contract_address, Bytes::from(vec_felt_to_bytes(calldata).0), BlockId::from(BlockNumberOrTag::Latest))
+            .await;
+
+        let token_allowance = match handle {
+            Ok(call) => {
+                let allowance = U256::try_from_be_slice(call.as_ref())
+                    .ok_or(ConversionError::<()>::Other("error converting from Bytes to U256".into()))
+                    .unwrap();
```

Comment:
can we remove this unwrap?

Reply:
Could we try with a "?" question mark
And then try to handle the error conversion as best we can (map_err being last resort)

Reply:
I managed to make the following work:

```rust
let allowance = U256::try_from_be_slice(call.as_ref())
    .ok_or(EthApiError::<P::Error>::ConversionError("error converting from Bytes to U256".into()))?;
```